runtime/jcl/common/jcldefine.c

```diff
@@ -181,6 +180,9 @@ defineClassCommon(JNIEnv *env, jobject classLoaderObject,
 				} else
 #endif /* defined(J9VM_OPT_SNAPSHOTS) */
 				{
+#if defined(J9VM_OPT_SNAPSHOTS)
```

Comment:
This should be done without ifdef. Remember, the original code wasn't assigning `clazz`.

Reply:
Actually, this doesnt work either since there are multiple paths. I think you should just revert back to the original code and add the flag `J9_FINDCLASS_FLAG_DONT_SKIP_FROZEN_JCL_DEFINE_CLASS`